syserol/regression.py

```diff
@@ -62,6 +63,24 @@ def noCMTF_function_prediction(tensorFac, function="ADCC"):
 
     return Y, Y_pred, np.sqrt(r2_score(Y, Y_pred))
 
+def SVR_noCMTF_function_prediction(tensorFac, function="ADCC"):
+    """ Predict functions using our decomposition and SVR regression methods"""
+    func, _ = importFunction()
+    df = pd.DataFrame(tensorFac[1][0])  # subjects x components matrix
+    df = df.join(func, how="inner")
+    df = df.dropna()
+    df_func = df[functions]
+    df_variables = df.drop(
+        ["subject"] + functions, axis=1
+    )
+
+    X = df_variables
+    Y = df_func[function]
+    Y_pred = cross_val_predict(SVR(kernel="linear"), X, Y, cv=10)
```

Comment:
```suggestion
    Y_pred = cross_val_predict(SVR(), X, Y, cv=len(Y))
```

The default `rbf` kernel would be better.

Reply:
Originally I did have it as rbf but for some reason it seemed to actually do better with linear? I can switch it though and see if it changes the figure much.

Reply:
That's a reasonable choice to make. Why I'd like to switch it to rbf even if it does worse, is the linear kernel is essentially the same thing as elastic net.